File: NICOLLE_DE_AVILA/nicolle_de_avila_peluffo/trabajos_en_ruby/juegopiedrapapeltijera.rb
opciones = ['Piedra', 'Papel', 'Tijeras']

while true
  puts 'Elige Piedra, Papel o Tijeras (o salir para terminar): '
  eleccion_usuario = gets.chomp

  break if eleccion_usuario.downcase == 'salir'

  indice_computadora = rand(3)
  eleccion_computadora = opciones[indice_computadora]

  puts "La computadora eligió: #{eleccion_computadora}"

  if eleccion_usuario == eleccion_computadora
    puts '¡Empate!'
  elsif (eleccion_usuario == 'Piedra' && eleccion_computadora == 'Tijeras') ||
        (eleccion_usuario == 'Papel' && eleccion_computadora == 'Piedra') ||
        (eleccion_usuario == 'Tijeras' && eleccion_computadora == 'Papel')
    puts '¡perdiste!'
  else
    puts '¡ganaste!'
  end
end

puts 'Gracias por jugar!'

File: NICOLLE_DE_AVILA/nicolle_de_avila_peluffo/trabajos_en_dart/juegopiedrapapeltijera.dart
import 'dart:io';
import 'dart:math';

void main() {
  List<String> opciones = ['Piedra', 'Papel', 'Tijeras'];
  Random random = Random();

  while (true) {
    stdout.write('Elige Piedra, Papel o Tijeras (o salir para terminar): ');
    String? eleccionUsuario = stdin.readLineSync();

    if (eleccionUsuario!.toLowerCase() == 'salir') {
      break;
    }

    int indiceComputadora = random.nextInt(3);
    String eleccionComputadora = opciones[indiceComputadora];

    print('La computadora eligió: $eleccionComputadora');

    if (eleccionUsuario == eleccionComputadora) {
      print('Empate!');
    } else if ((eleccionUsuario == 'Piedra' && eleccionComputadora == 'Tijeras') ||
               (eleccionUsuario == 'Papel' && eleccionComputadora == 'Piedra') ||
               (eleccionUsuario == 'Tijeras' && eleccionComputadora == 'Papel')) {
      print('¡perdiste!');
    } else {
      print('¡ganaste!');
    }
  }

  print('Gracias por jugar!');
}